### PR TITLE
chore(dot/network): improve CPU usage again; no more mallocs

### PR DESCRIPTION
**Pull request by noot**
## Changes

- update `readLEB128ToUint64` to read into existing buffer 
- reduce `maxMessageSize` to 63kb since messages weren't actually using up the size allocated before (1mb)
- move connect to peers if below minimum case to its own goroutine, only find peers if below min

## Tests

```
go test ./dot/network -short
```

## Issues

- related to #1494

### CODE REVIEW

**Review by EclesioMeloJunior — APPROVED** [review]

File: dot/network/discovery.go [hunk 1]
```diff
@@ -35,13 +35,15 @@ var (
 	startDHTTimeout             = time.Second * 10
 	initialAdvertisementTimeout = time.Millisecond
 	tryAdvertiseTimeout         = time.Second * 30
-	connectToPeersTimeout       = time.Minute
+	connectToPeersTimeout       = time.Minute * 5
+	findPeersTimeout            = time.Minute
```
Comment (timwu20) [inline dot/network/discovery.go:39]: nit: I prefer to put these configurable settings inside the struct, so `discovery` in this case.

File: dot/network/service.go [hunk 2]
```diff
@@ -139,10 +139,10 @@ func NewService(cfg *Config) (*Service, error) {
 	var bufPool *sizedBufferPool
 	if cfg.noPreAllocate {
 		bufPool = &sizedBufferPool{
-			c: make(chan *[maxMessageSize]byte, cfg.MaxPeers*3),
+			c: make(chan *[maxMessageSize]byte, cfg.MinPeers*3),
```
Comment (timwu20) [inline dot/network/service.go:142]: question: why *3?

**Review by timwu20 — APPROVED** [review]

File: dot/network/service.go (hunk 2 above)
Comment (noot) [reply dot/network/service.go:142]: there's currently 3 notifications protocols supported, so the number of open streams  is likely to be peercount * 3 (I think the comment explains this)

File: dot/network/discovery.go (hunk 1 above)
Comment (noot) [reply dot/network/discovery.go:39]: would you put them inside the struct then assign the value inside `newDiscovery`?

Comment (timwu20) [reply dot/network/discovery.go:39]: Yea then you can set your desired timeout in the tests per `*discovery` rather than globally through the package.

**Pull request merged**